handlers.go

```diff
@@ -48,34 +48,47 @@ func (h MethodHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// Logging
+
+// FormatterParams is the structure any formatter will be handed when time to log comes
+type FormatterParams struct {
+	Writer     io.Writer
```

Comment:
I didn't want to include the writer here, the goal wasn't to just put all the parameters into a struct but to separate "what is being logged" into just one field so that is extensible, and so similarly typed parameters like StatusCode and Size can't be accidentally transposed or misinterpreted. So I would put the writer back as a separate argument.

Also I'd prefix the name of this struct with `LogFormatter` because there are a lot of other handlers in this package apart from the logging handler, so it helps make it clear where it belongs.